app/src/main/java/com/example/rwblickhan/webviatextserver/MainActivity.java

```diff
@@ -1,13 +1,52 @@
 package com.example.rwblickhan.webviatextserver;
 
+import android.os.Handler;
+import android.os.Message;
+import android.os.Looper;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
 public class MainActivity extends AppCompatActivity {
 
+    private static final int RECEIVE_SMS = 1;
+    Handler mHandler = new SMSHandler();
+
+    static class SMSHandler extends Handler
+    {
+        public void handleMessage(Message msg)
+        {
```

Comment:
My message contains a hash map mapping an SMS address to the URL contained in the text, so that'll have to be dealt with.